File: _includes/code/search-a-2d-matrix/solution.py
from typing import List


class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n, m = len(matrix), len(matrix[0])
        row, col = 0, m - 1
        while row < n and col >= 0:
            cell = matrix[row][col]
            if cell == target:
                return True
            if cell < target:
                row += 1
            else:
                col -= 1
        return False

File: _includes/code/search-a-2d-matrix/solution.go
package search_a_2d_matrix

func searchMatrix(matrix [][]int, target int) bool {
	n := len(matrix)
	m := len(matrix[0])
	row := 0
	col := m-1
	for row < n && col >=0 {
		cell := matrix[row][col]
		if cell == target {
			return true
		}
		if cell < target {
			row++
		} else {
			col--
		}
	}
	return false
}
